raft/src/main/java/org/apache/kafka/raft/internals/KRaftControlRecordStateMachine.java

```diff
@@ -70,6 +71,9 @@ public final class KRaftControlRecordStateMachine {
     // 2. The read operations lastVoterSet, voterSetAtOffset and kraftVersionAtOffset read
     // the nextOffset first before reading voterSetHistory or kraftVersionHistory
     private volatile long nextOffset = STARTING_NEXT_OFFSET;
+    private final KafkaRaftMetrics kafkaRaftMetrics;
+    private final ExternalKRaftMetrics externalKRaftMetrics;
+    private final Optional<VoterSet> staticVoterSet;
```

Comment:
You can just model this with `VoterSet staticVoterSet`. `Optional.empty()` has the same meaning as `VoterSet.empty()` or `VoterSet#isEmpty()`.